### ISSUE TITLE
Question/Feature request: PNG metadata for ML-created images.

### ISSUE THREAD

**Issue by Hotcooler** [Inlustra/next-photo-stream#4]
First of all thanks for this great project, since it's indeed incredibly hard to find anything for a ready-to-go image display site, with basically a dump directory interface. So it's a godsend that this project exists.

A bit of pretext, I've recently got a new GPU, and... got into messing about with Stable Diffusion, and was becoming a bother to friends on messengers sending them pics ;) So I kinda thought a small simple gallery site that just displays what is dumped into a directory with some simple modify-date sort must be easy to come by :) Apparently it aint haha.

As a fellow connoisseur of AI generated imagery, it would be rather cool to add the ability to display creation parameters when viewing in full screen under the picture. SD saves it's image generation parameters in a tEXt field inside a PNG with a key=parameters.

As I imagine I might not be the only one looking at this project to display ML images to friends, it might be a cool addition. Plus probably the same can be used for displaying relevant (date, place e.t.c.) EXIF for your wedding photos too :)

**Comment by Inlustra** [comment]
So I just looked into this, this data isn't in the EXIF as you've mentioned, but I can't see exactly how to get this `tEXt field` do you have any docs on it I can look at? 

That being said, I don't have lots of time to spend on this and I don't think I can commit to writing any image file processing. 

We do already pull the EXIF, but from my brief attempts at least, I don't see it in there.

**Comment by Hotcooler** [comment]
Yeah it's not a priority for sure, it's just a if you have time/want for this "nice to have".

It's not EXIF, but I imagine libraries that do pull EXIF can also do png metadata fields. Here's a description from EXIV2 - https://dev.exiv2.org/projects/exiv2/wiki/The_Metadata_in_PNG_files

I figured IMAGOR might already be able to extract that data (though afaik only EXIF is mentioned). Tested, Exiftool can get that, imagemagik's identify can do it,  pngcheck -7 too, exiv2 from ubuntu repo - cannot.

I'll attach an image to look at.
![00002-3232408443-photo of 8k u](https://user-images.githubusercontent.com/4125481/197783283-038bc2d2-8967-4788-be59-0c45e562ce54.png)
Here's what it looks like inside, basically plain text after the header.
![Clipboard02](https://user-images.githubusercontent.com/4125481/197784736-c967cb64-b644-4fe4-b941-4164c7ea2b36.png)
![Clipboard03](https://user-images.githubusercontent.com/4125481/197784739-a1ae5e8a-3fa5-48e9-b235-c46dec449823.png)
